tests/integration_tests.py

```diff
@@ -647,6 +647,350 @@ def test_will_not_output_empty_state(self):
                           {"bookmarks":{"chicken_stream":{"id": 1 }},
                            'currently_syncing' : 'chicken_stream'})
 
+class BufferingPerStreamConstraints(unittest.TestCase):
```

Comment:
This needs state assertions for the happy paths

Reply:
Done.